geoextent/lib/extent.py

```diff
@@ -0,0 +1,129 @@
+    input "filePath": type string, path to file from which the metadata shall be extracted \n
+    input "whatMetadata": type string, specifices which metadata should be extracted  \n
+    returns None if the format is not supported, else returns the metadata of the file as a dict 
+    (possible) keys of the dict: 'temporal_extent', 'bbox', 'vector_reps', 'crs'
+    '''
+    logging.info("Extracting {} from file {}\n".format(whatMetadata, filePath))
+    
+    fileFormat = filePath[filePath.rfind('.')+1:]
+    usedModule = None
+
+    # initialization of later output dict
+    metadata = {}
+
+    # first get the module that will be called (depending on the format of the file)
+    if fileFormat == 'geojson' or fileFormat == 'json':
+        usedModule = handleGeojson
+    elif fileFormat == 'csv':
+        usedModule = handleCSV
+    elif fileFormat == 'shp' or fileFormat == 'dbf':
+        usedModule = handleShapefile
+    elif fileFormat == 'geotiff' or fileFormat == 'tif':
+        usedModule = handleGeotiff
+    else: 
+        # file format is not supported
+        return None
+    #only extracts metadata if the file content is valid
+    try:
+        fileValidity = usedModule.checkFileValidity(filePath)
+    except Exception as e:
+        print("Error for " + filePath + ": " + str(e))
+        fileValidity = 'notValid'
+
+    #get Bbox, Temporal Extent, Vector representation and crs parallel with threads
+    class thread(threading.Thread): 
+        def __init__(self, thread_ID): 
+            threading.Thread.__init__(self) 
+            self.thread_ID = thread_ID
+        def run(self):
+            metadata["format"] = usedModule.fileType
+            #print("Thread with Thread_ID " +  str(self.thread_ID) + " now running...")
+            #metadata[self.thread_ID] = self.thread_ID
+            if self.thread_ID == 100:
+                try:
+                    metadata["bbox"] = computeBboxInWGS84(usedModule, filePath)
+                except Exception as e:
+                    print("Warning for " + filePath + ": " + str(e)) 
+            elif self.thread_ID == 101:
+                try:
+                    metadata["temporal_extent"] = usedModule.getTemporalExtent(filePath)
+                except Exception as e:
+                    print("Warning for " + filePath + ": " + str(e))
+            elif self.thread_ID == 103:
+                try:
+                    # the CRS is not neccessarily required
+                    if hasattr(usedModule, 'getCRS'):
+                        metadata["crs"] = usedModule.getCRS(filePath)
+                    else: 
+                        print ("Warning: The CRS cannot be extracted from the file")
+                except Exception as e:
+                    print("Warning for " + filePath + ": " + str(e))      
+            try:
+                barrier.wait()
+            except Exception as e:
+                barrier.abort()
+
+    thread_bbox_except = thread(100) 
```

Comment:
Does this mean that the function will always use three threads to extract bbox, temporal extent, and CRS? If so, we must document that, at least in the function documentation.